tests/python_api/tf_detection_reader.py

```diff
@@ -48,17 +48,13 @@ def get_weights(num_bboxes):
     return weights_array
 
 
-def draw_patches(img, idx, bboxes, device_type, args=None):
+def draw_patches(img, idx, bboxes, device, args=None):
     import cv2
     args = parse_args()
-    if device_type == "gpu":
-        try:
-            import cupy as cp
-            img = cp.asnumpy(img)
-        except ImportError:
-            pass
     if not args.NHWC:
         img = img.transpose([0, 1, 2])
+    if device == "gpu":
```

Comment:
@LakshmiKumar23 Same issue as TF classification reader example. Please move the device check to L54